### ISSUE TITLE
Yarn after turbo prune -> Syntax Error: Expected "\n", "\r", or "\r\n" but ":" found.

### ISSUE THREAD

**Issue by filipjakov** [vercel/turborepo#1975]
### What version of Turborepo are you using?

1.4.6

### What package manager are you using / does the bug impact?

Yarn v2/v3 (node_modules linker only)

### What operating system are you using?

Mac

### Describe the Bug

After running:

```sh
turbo prune --scope=<workspace>
```

i enter the generated `out` folder. Running any yarn command in that folder, results in the following error:

```
Syntax Error: Expected "\n", "\r", or "\r\n" but ":" found.
    at ec (/Users/name/.node/corepack/yarn/3.2.3/yarn.js:45:1124)
    at Mpe (/Users/name/.node/corepack/yarn/3.2.3/yarn.js:45:9766)
    at Kpe (/Users/name/.node/corepack/yarn/3.2.3/yarn.js:327:16)
    at jpe (/Users/name/.node/corepack/yarn/3.2.3/yarn.js:327:114)
    at Si (/Users/name/.node/corepack/yarn/3.2.3/yarn.js:327:490)
    at ze.setupResolutions (/Users/name/.node/corepack/yarn/3.2.3/yarn.js:439:363)
    at async Function.find (/Users/name/.node/corepack/yarn/3.2.3/yarn.js:436:1533)
    at async bm.execute (/Users/name/.node/corepack/yarn/3.2.3/yarn.js:572:1845)
    at async bm.validateAndExecute (/Users/name/.node/corepack/yarn/3.2.3/yarn.js:345:673)
    at async Bs.run (/Users/fbulic/.node/corepack/yarn/3.2.3/yarn.js:359:2087)
```

After copying the global `yarn.lock` and replacing the generated one, everything runs as expected.

### Expected Behavior

Yarn should work as expected with prune/in the generated out folder.

### To Reproduce

Environment:
- node: 16.17.0
- yarn (in corepack env): 3.2.3 (locally installed)

**Comment by chris-olszewski** [comment]
Hi @filipjakov, currently prune support for Yarn 2+ is a "best attempt" and usually doesn't work. I'm currently working on a correct implementation. See #643 for more discussion about Yarn 2+ support

**Comment by chris-olszewski** [comment]
With #2019 merged and 1.5 released this should be solved. Please update turbo and try this again. Reopen if the issue persists.
